firestore.rules

```diff
@@ -54,19 +54,32 @@ service cloud.firestore {
     }
 
     match /users/{userId} {
-      function validateAttrs(){
-        return request.resource.data.keys().hasOnly(['displayName', 'avatar'])
+      function validateDisplayAndName(){
+        return request.resource.data.keys().hasOnly(['displayName', 'avatar', 'catWins', 'catfishWins', 'playedAsCat', 'playedAsCatfish'])
           && request.resource.data.displayName is string && request.resource.data.displayName.size() >= 3 
           && request.resource.data.displayName.size() <= 15 
           && request.resource.data.displayName == request.resource.data.displayName.trim()
           && request.resource.data.displayName.matches('[A-Za-z0-9-_ ]+')
           && request.resource.data.avatar is number && int(request.resource.data.avatar) == request.resource.data.avatar 
           && request.resource.data.avatar >= 0 && request.resource.data.avatar <=12;
       }
+      function validateStats() {
+          return request.resource.data.keys().hasOnly(['displayName', 'avatar', 'catWins', 'catfishWins', 'playedAsCat', 'playedAsCatfish'])
+          && request.resource.data.catWins == 0 && request.resource.data.catfishWins == 0
+          && request.resource.data.playedAsCat == 0 && request.resource.data.playedAsCatfish == 0;
+      }
+      function validateOnUpdate(){
```

Comment:
might be helpful to say what this function is doing. maybe you could call it statsAreNotChanged